patterns/1-initial/extensions-to-manage-contributions-at-scale.md

```diff
@@ -0,0 +1,81 @@
+## Title
+
+Extensions to Manage Contributions at Scale
+
+## Patlet
+
+TBD
+
+## Problem
+
+As the number of new feature contributions to a mature inner-source repository rapidly increases, it adds more burden on code reviews and maintenance. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping repositories well maintained?
+
+## Context
+
+- More employees are contributing to strategic inner-source codebases, some of which are scaling rapidly with new feature contributions. This is adding a huge code review backlog on a smaller group of maintainers, resulting in maintainer burnout as well as slowing down release of new capabilities.
+- Adding an excessive number of capabilities and code to repositories is making them difficult to maintain.
+- here is a growing backlog of new features and ideas for the products which have scaled.
+- The pattern applies in either scenario:
+   - Maintainers find themselves rejecting new feature ideas to narrow down the scope of the products. This is hampering innovation in the community and restricting further expansion.
+   - To reduce backlog, new features are getting released without thorough documentation, hardening, or testing, creating a poor user experience. This is also bloating the size of the codebases, adding a huge dependency graph and making them difficult to maintain.
+
+## Forces
+
+- Maintainers and product owners want to allow for expansion, encourage innovation and experimentation without being very restrictive on contributions, while also keeping good code and quality standards for user experience.
+- A large amount of time goes into hardening and thorough testing of features to meet product standards, but product owners may want to allow for faster release of new innovations for adopting products to explore before investing time in productionization of capabilities.
+- Maintainers want to encourage the community to share innovations that combine product capabilities with other use-cases without adding more dependencies to the primary repository.
+
+## Solutions
+
+Allowing extensions/plugins to high-scale inner-source codebases or libraries can relieve the maintenance burden on repository maintainers and allow faster release of new features for adopting products to explore. This shifts maintenance of capabilities to extension owners and allows the primary repository to support capabilities that have been adopted more widely and are more strategic. Extensions provide a filter for new capabilities that may eventually move into the core of the project. Extensions also act as an incubation and community hardening environment, allowing for much of that hardening to happen organically rather than in a costly review process.
```

Comment:
```suggestion
Allowing extensions/plugins to high-scale InnerSource codebases or libraries can relieve the maintenance burden on repository maintainers and allow faster release of new features for adopting products to explore. This shifts maintenance of capabilities to extension owners and allows the primary repository to support capabilities that have been adopted more widely and are more strategic. Extensions provide a filter for new capabilities that may eventually move into the core of the project. Extensions also act as an incubation and community hardening environment, allowing for much of that hardening to happen organically rather than in a costly review process.
```